src/encoding.md

```diff
@@ -344,6 +345,57 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-invalid-csv]: #ex-invalid-csv
+<a name="ex-invalid-csv"></a>
+## Handle invalid CSV data with Serde
+
+[![csv-badge]][csv] [![serde-badge]][serde] [![cat-encoding-badge]][cat-encoding]
+
+CSV files often contain invalid data. For these cases the csv crate
+provides a custom deserializer, [`csv::invalid_option`], which automatically
+converts invalid data to None values.
+
+[`csv::invalid_option`]: https://docs.rs/csv/*/csv/fn.invalid_option.html
+
+```rust
+# #[macro_use]
+#extern crate error_chain;
+extern crate csv;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+
+#[derive(Debug, Deserialize)]
+struct Record {
+    name: String,
+    #[serde(deserialize_with = "csv::invalid_option")]
+    place: Option<String>,
+    id: u64,
+}
+#
+# error_chain! {
+#     foreign_links {
+#          CsvError(csv::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let data = "name,place,id
+                mark,\"\",46
```

Comment:
If we leave out one field, it will actually give error of fields: "found record with 2 fields, but the previous record has 3 fields".